packages/core/src/crypto/jose/jwk/Jwk.ts

```diff
@@ -0,0 +1,40 @@
+import type { Buffer } from '../../../utils'
+import type { KeyType } from '../../KeyType'
+import type { JwaKeyType, JwaEncryptionAlgorithm, JwaSignatureAlgorithm } from '../jwa'
+
+import { Key } from '../../Key'
+
+export interface JwkJson {
+  kty: string
+  use?: string
+  [key: string]: unknown
+}
+
+export abstract class Jwk {
+  public abstract publicKey: Buffer
+  public abstract keyType: KeyType
+  public abstract supportedSignatureAlgorithms: JwaSignatureAlgorithm[]
+  public abstract supportedEncryptionAlgorithms: JwaEncryptionAlgorithm[]
+
+  public abstract kty: JwaKeyType
+  public use?: string
+
+  public toJson(): JwkJson {
```

Comment:
Would it not make more sense to make this abstract?

Reply:
Well this covers the base of the JWK. For now we only support two fields, but we can extend this with more over time (and even add `additionalFields` that will be added). So I think having a base is nice, to not require all extensions to do the same